CorsixTH/Src/th_movie.cpp

```diff
@@ -136,9 +116,8 @@ THMoviePictureBuffer::~THMoviePictureBuffer()
 void THMoviePictureBuffer::abort()
 {
     m_fAborting = true;
-    SDL_LockMutex(m_pMutex);
-    SDL_CondSignal(m_pCond);
-    SDL_UnlockMutex(m_pMutex);
+    std::lock_guard<std::mutex> lock(m_mutex);
+    m_cond.notify_all();
```

Comment:
Intentional - releasing should unlock every thread waiting for the queue. It's theoretical though because currently only one other thread would be waiting.